scala-package/core/src/main/scala/org/apache/mxnet/FeedForward.scala

```diff
@@ -230,8 +230,11 @@ class FeedForward private(
       val padded = batch.pad
       val realSize = batchSize - padded
       for ((list, nd) <- outputs zip predExec.outputs) {
-        list += nd.slice(0, realSize).copy()
+        val ndSliced = nd.slice(0, realSize)
+        list += ndSliced.copy()
+        ndSliced.dispose()
       }
+      batch.dispose()
```

Comment:
Same comment as above about using try/finally to make sure dispose always happens, even in case of an exception.